services/streammanager/kafka/kafkamanager.go

```diff
@@ -450,14 +450,19 @@ func serializeAvroMessage(value []byte, codec goavro.Codec) ([]byte, error) {
 	return binary, nil
 }
 
-func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timestamp time.Time, p producerManager) (
+func prepareBatchOfMessages(batch []map[string]interface{}, timestamp time.Time, p producerManager, defaultTopic string) (
```

Comment:
Instead of passing the `defaultTopic` around, an alternative approach is to use the existing pattern of including such information in the `producerManager`:

```go
type producerManager interface {
	io.Closer
	publisher
	getDefaultTopic() string
	getTimeout() time.Duration
	getCodecs() map[string]*goavro.Codec
}
```

That should be preferred given that it is already in place. I've attached a git diff for you to consider this approach (e.g. `git apply yourcoworkers.diff`).

[kafka.diff.zip](https://github.com/rudderlabs/rudder-server/files/9950299/kafka.diff.zip)

Reply:
Can we take this up in the next iteration of enhancement I can create a backlog item in our board for this. The reason I am asking is I need to update and add some test cases for this approach as we are dangerously close to release this might take more time :/

Reply:
Sure no problem :+1: